### ISSUE TITLE
Increase the duration of validity of the receipt.

### ISSUE THREAD

**Issue by nickdesaulniers** [mozilla/r2d2b2g#497]
Increase the [duration of validity](https://bugzilla.mozilla.org/show_bug.cgi?id=867662) of the receipt.

**Comment by mykmelez** [comment]
@nickdesaulniers: do you plan to work on this in the near future? If so, let's assign this to you. Otherwise, let's move it out of the v4 milestone. Your pick!

**Comment by mykmelez** [comment]
We need to keep valid receipts valid, so users don't have to manually deal with an expired receipt. Increasing their duration of validity might mitigate the problem, but it won't eliminate it (unless we can increase it to infinity!).

@nickdesaulniers Perhaps we can morph this issue into "keep valid receipts valid" and use it to track the work for that enhancement?

**Comment by nickdesaulniers** [comment]
@andymckay Would it be possible to have test receipts have a far future expiration date, since developers should be screening for them anyways?  So maybe instead of adding the query string parameter to the request, and the additional UI to the simulator, just always fetch a long lived valid receipt?

**Comment by andymckay** [comment]
You filed a bug about that I just made it very low priority because its pointless until we've got the rest of the infrastructure in place. We are blocked on the platform with https://bugzilla.mozilla.org/show_bug.cgi?id=757226, once thats cleared, we can start to worry about dealing with receipt expiry.

**Comment by nickdesaulniers** [comment]
@andymckay , sure but now an app installed with a valid receipt in the simulator will be invalid after 24 hours.  Instead of having the dev have to reinstall the app every day, why can't the VALID test receipts have a far future expiration date?  Isn't the expiration time something that is changeable when it's generated?  Why is it currently set to 24 hours?  I'd be happy to close [bug 867662](https://bugzilla.mozilla.org/show_bug.cgi?id=867662) (having the API to choose the expiration) in exchange for just getting VALID receipts that default to something longer than 24 hours.  I agree having [bug 757226](https://bugzilla.mozilla.org/show_bug.cgi?id=757226) land will be helpful, but I don't think work on simulator receipt support should block on it.  Maybe there is another way we can solve the need-to-reinstall-every-24 hour problem?

**Comment by andymckay** [comment]
The expiration time is changeable, however nothing checks it because you cannot replace a receipt. The simulator should not be checking the expiration date, just like apps should not be checking the expiration receipt.

**Comment by nickdesaulniers** [comment]
Ah, so I should just remove the option from the simulator to install an expired receipt, and we should remain aware that once [bug 757226](https://bugzilla.mozilla.org/show_bug.cgi?id=757226) has been patched, and the receipt validator flags expired receipts, then we should worry about durations.  For now, this is non issue.  So the fix for this will be remove the option for installing receipts.  @mykmelez and I also discussed implementing a fake receipt verification service, should the need arise.

**Comment by andymckay** [comment]
Right, once receipt refreshes have landed, we can all start worrying about expiry and use things like https://wiki.mozilla.org/Apps/WebApplicationReceiptRefresh. For the moment the validator will tell you that the receipt is valid, just expired and everything would ignore it. I'm not sure why you'd want a fake receipt verification service, the marketplace is already one.

**Comment by nickdesaulniers** [comment]
fixed in #551
